### ISSUE TITLE
[clang-tidy] Add AllowedTypes option to misc-const-correctness

### ISSUE THREAD

**Issue by vbvictor** [llvm/llvm-project#122592]
Sometimes in large codebases we want to ignore const-correctness of certain classes because they are immutable by nature, so there is no point in adding const to them.
Minimal example:

```cpp
#include <mutex>

int main() {
  std::mutex m;
  std::lock_guard<std::mutex> l(m); // check is triggered but we want to ignore it since lock_guard is already immutable.
}
```

Writing NOLINT for thousands of places bloats code and is not an appealing option.

**Comment by vbvictor** [comment]
Can this issue be assigned to me if it is a good improvement

**Comment by firewave** [comment]
There is #58084 which is somewhat related by adding an option to only report it for POD types.

**Comment by vbvictor** [comment]
> There is https://github.com/llvm/llvm-project/issues/58084 which is somewhat related by adding an option to only report it for POD types.

This option may help with my particular case with `std::lock_guard` however, I think narrowing to POD types may be too strict for some users, including me.

**Comment by firewave** [comment]
> This option may help with my particular case with `std::lock_guard` however, I think narrowing to POD types may be too strict for some users, including me.

See all my notes in the other ticket what the actual intention is. The title is a bit misleading.
